dsc/utilities/aws/sqs.py

```diff
@@ -139,17 +151,35 @@ def send(
             message_attributes: The attributes of the message to send.
             message_body: The body of the message to send.
         """
-        logger.debug("Sending message to SQS queue: %s", self.queue_name)
+        logger.debug(f"Sending message to SQS queue: {self.queue_name}")
         response = self.client.send_message(
-            QueueUrl=f"{self.base_url}{self.queue_name}",
+            QueueUrl=self.queue_url,
             MessageAttributes=message_attributes,
             MessageBody=str(message_body),
         )
-        logger.debug("Response from SQS queue: %s", response)
+        logger.debug(f"Response from SQS queue: {response}")
         return response
 
+    def validate_message(self, sqs_message: MessageTypeDef) -> bool:
```

Comment:
I know that I submitted an approval already, and I do think this current approach will successfully validate messages, but taking another pass at the updates I do wonder if there is an opportunity to streamline some of this validation even more.  I would imagine that _much_ of this application will be about validation, so some patterns to follow throughout could be helpful to establish early.

I see that all this validation is under the umbrella of [this call to self.validate_message()](https://github.com/MITLibraries/dspace-submission-composer/blob/IN-1099-ses-and-sqs-clients/dsc/utilities/aws/sqs.py#L117-L118) in `self.process_result_message()`:

```python
if not self.validate_message(sqs_message):
    raise InvalidSQSMessageError
```

Where the ultimate goal is to a) log what happend, and b) raise `InvalidSQSMessageError` from this method.

What if each validation helper method was responsible for directly raising `InvalidSQSMessageError` if they found something wrong?  and `self.validate_message()` is just an orchestrator of calling these methods?  Something along the lines of:

```python
def self.validate_message() -> None:
    # call all validators here
    # if all pass, return None, and all is well!
    # if any raise an exception, it will bubble up from here and reach self.process_result_message()
    #   where this was originally called from

def validate_message_attributes() -> None:
   # each validator returns None
   # do checks, raise InvalidSQSMessageError with a meaningful message if something wrong

# continue to define validators as needed
```

I think this leans into the EAFP (Easier to ask for forgiveness than permission) python pattern.  If we call `self.validate_message()` and nothing happens, great.  But if any of those validator methods raise that `InvalidSQSMessageError` exception, it will:

1. get raised from that lowest validator method
2. bubble up through `validate_message()`
3. bubble up from `process_results_message()` [here](https://github.com/MITLibraries/dspace-submission-composer/blob/IN-1099-ses-and-sqs-clients/dsc/utilities/aws/sqs.py#L117) just as it does now

This pattern is extensible, becuase you can just keep calling validator methods, knowing they are responsible for raising that exception.

To reiterate: with only two validations here, I think the current approach works as-is.  But I have this hunch that this DSC app will be validation heavy, and a validation pattern that is applied somewhat globally could be helpful.

Reply:
I like @ghukill 's proposal to have `validate_message` simply call validation methods that are responsible for raising `InvalidSQSMessageError` with meaningful messages.

Reply:
Thank you, this is definitely a better approach, I greatly appreciate the suggestion and I'll rework!